src/evaluation/retrieval.py

```diff
@@ -1,80 +1,106 @@
 import open_clip
 import numpy as np
 import torch
+from einops import rearrange
 
-
-def retrieval_evaluation(model_video, model_text, data, multicaption=False):
+def retrieval_evaluation(model_video, model_text, data):
     if type(data) == dict:
         dataloader = data["val"].dataloader
     else:
         dataloader = data
     device = "cuda" if torch.cuda.is_available() else "cpu"
     all_video_features, all_text_features = [], []
-    ground_truth = []
-    samp = 0
-
+    max_txt_len = 1
+    
     with torch.no_grad():
         for i, batch in enumerate(dataloader):
             embeddings = batch["embeddings"]
             toks = []
             # TODO: does this require batch_size = 1 ??
             for cap in batch["text"]:
-                if multicaption:
-                    for c in cap.split(";"): # multiple captions separated by ;
-                        toks.append(open_clip.tokenize(c))
-                        ground_truth.append(samp)
-                else:
-                    toks.append(open_clip.tokenize(cap))
-                    ground_truth.append(samp)
-                samp += 1
+              caps_list = cap.split(';')
```

Comment:
why aren't we checking for the 'multicaption' param? maybe some datasets have one caption but the caption has ';' in it?